readthedocs/donate/templates/donate/promo_detail.html

```diff
@@ -11,25 +11,59 @@
 
 {% if promos %}
 
+{% if promos|length > 1 %}
 <p>
-Total Clicks for all shown promos: {{ total_clicks }}
+<strong>Total Clicks for all shown promos</strong>: {{ total_clicks }}
 </p>
+{% endif %}
 
+<div id="promo_detail">
 {% for promo in promos %}
 
 <h3>
 Results for {{ promo.name }} ({{ promo.analytics_id }}) over last {{ days }} days.
 </h3>
 
 <div class="example" style="width: 30%;">
-<a href="{{ promo.link }}"><img width=120 height=90 src="{{ promo.image }}"></a>
-<br>
-{{ promo.text|safe }}
 
-</div>
+  <div class="filters">
+    {% if promo.programming_language %}
+      <p> 
+        <strong>Filtered Language</strong>: {{ promo.programming_language }}
+      </p>
+    {% endif %}
+
+    {% if promo.geo_filters.count %}
+      <p> 
+      <strong>Filtered Geos</strong>: 
+        {% for geo in promo.geo_filters.all %}
+          {{ geo.countries.all|join:", " }}
+        {% endfor %}
+      </p>
+    {% endif %}
+
+    {% if promo.sold_clicks %}
+      <p> 
+        <strong>Total Clicks Sold</strong>: {{ promo.sold_clicks }}
+      </p>
+    {% endif %}
+  </div>
+
+  <a href="{{ promo.link }}">
+    <img width=120 height=90 src="{{ promo.image }}">
+  </a>
+
+  <br>
+
+  <div class="promo_text">
+    {{ promo.report_html_text|safe }}
+  </div>
+
+  <br>
```

Comment:
So, this is not necessarily an issue that requires changes, but the main problem with our UI and templates is lack of consistency, so I'll point out that:

* The styling you are looking for is closer to a `<dl>`, instead of recreating this with a paragraph tag
* `<strong>` isn't used semantically here. The `<strong>` text here is structural and should be handled with a general list that styles the label and value in a generic way.
* We should never need `<br>` in our templates, whitespace should be left to CSS. We can't remove `<br>` using CSS, so this makes for noise when restyling.

This is just informational. I'm still planning out what a UI overhaul will be, I'd like to stop collecting debt here in the meantime though.